tools/src/h5repack/h5repack_main.c

```diff
@@ -875,6 +886,23 @@ parse_command_line(int argc, const char *const *argv, pack_opt_t *options)
         ret_value = -1;
     }
 
+    /* If the input file uses the onion VFD, get the revision number */
+    if (in_vfd_info.u.name && !HDstrcmp(in_vfd_info.u.name, "onion")) {
```

Comment:
The idea behind the --vfd-info options was to be able to pass arbitrary info to a VFD and make the VFD responsible for handling the info object. I'm not super excited about adding more custom handling for a specific VFD into each HDF5 tool; it gets hard to correctly maintain and the tools probably shouldn't be aware of particular VFDs in the first place.

Reply:
Let's talk and figure out a good solution.